db/sqlinterfaces.c

```diff
@@ -2380,6 +2380,7 @@ int handle_sql_commitrollback(struct sqlthdstate *thd,
 
     if (rc == SQLITE_OK) {
         /* send return code */
+        clnt->has_recording = 0;
```

Comment:
This is too early; we need to capture rc !=SQLITE_OK as well.  
For an more robust cleanup, in case of multi-statement transaction, I recommend cleaning up the flag at a higher level (in handle_non_sqlite_request, which captures begin, wrong-state, and commit-rollback case).